### PR TITLE
Sync refactor tweaks

### LINKED ISSUE #3496: Importer progress bars not updating when in background
**Describe the bug**
In testing #3285, I found that if the import completes while the browser is not the foreground window, the progress bars are not updated to show that the import is complete. This occurs on both that PR and master.

**To Reproduce**

1. Deactivate WC Admin
2. Create 500 orders with smooth generator
3. Activate WC Admin
4. Start the import & stay on the settings screen until the progress bars show the import in progress
5. Put the Settings screen in the background & wait for the import to complete
6. Bring Settings to foreground & see that the message is updated but the progress bar(s) are not.

**Screenshots**
Screen Shot 2019-12-31 at 6 57 50 PM

**Desktop (please complete the following information):**

* OS: MacOS
* Browser: Chrome
* Version Latest

### PR DESCRIPTION
**Pull request by rrennick**
Fixes #3496 

This is a followup PR to #3285. With AS 2.2.X, the probability of the import status api call occurring while actions are in progress is about 5% (3/60). With AS 3.0, that probability is much closer to 100% when the last import actions are processing. This PR adds a check for `in-progress` actions if there are no remaining `pending` actions.

### Detailed test instructions:

- Activate Action Scheduler 3.0
- Delete existing imported analytics data
- Import analytics data
- Remain on screen to verify that 
- The `Finalizing ...` message appears
- The import ends with the message `Historical data from onward imported`

### CODE REVIEW

**Comment by jeffstieler** [comment]
@rrennick looks like this is bringing a lot of commits that were squashed in the merge to master. Can you rebase or cherry pick just your new commits?

**Comment by rrennick** [comment]
> Can you rebase or cherry pick just your new commits?

Cherry picked.

File: src/Schedulers/ImportScheduler.php [hunk 1]
```diff
@@ -38,8 +38,20 @@ public static function is_importing() {
...
+					'search'   => 'import',
+					'group'    => self::$group,
+				)
+			);
+		} else {
+			$in_progress = true;
+		}
 
-		return ! empty( $pending_jobs );
+		return ! empty( $in_progress );
```
Comment (joshuatf) [inline src/Schedulers/ImportScheduler.php]: The logic with the if/else above was a little hard to follow.  Maybe we could make this:

```php
return ! empty( $pending_jobs ) || ! empty( $in_progress_jobs );
```

File: src/Schedulers/SchedulerTraits.php [hunk 2]
```diff
@@ -234,7 +234,11 @@ public static function get_next_blocking_job( $action_name ) {
 		if ( is_array( $blocking_jobs ) ) {
 			foreach ( $blocking_jobs as $blocking_job ) {
 				$blocking_job_hook = $blocking_job->get_hook();
-				$next_job_schedule = $blocking_job->get_schedule()->next();
+				if ( method_exists( $blocking_job->get_schedule(), 'get_date' ) ) {
```
Comment (joshuatf) [inline src/Schedulers/SchedulerTraits.php]: Curious why we use `get_date()` here- is `next()` not available to in progress actions?  And is `get_date()` not available to pending actions?

Comment (rrennick) [reply src/Schedulers/SchedulerTraits.php]: `next()` logs a deprecated warning in 3.0. `get_date()` was added in 3.0.

Comment (joshuatf) [reply src/Schedulers/SchedulerTraits.php]: That makes sense - thanks for the explanation.

File: src/Schedulers/ImportScheduler.php (hunk 1 above)
Comment (joshuatf) [reply src/Schedulers/ImportScheduler.php]: Not sure if you've finished updating or not, but I was thinking we remove the latter else portion:

```
else {
	$in_progress = true;
}
```

In favor of checking `$pending_jobs` and `$in_progress_jobs`.

Comment (rrennick) [reply src/Schedulers/ImportScheduler.php]: @joshuatf Thanks for the followup. Removed in `0d42422`.
